source/aws-bootstrap-kit/lib/organization.ts

```diff
@@ -14,145 +14,104 @@ See the License for the specific language governing permissions and
+   */
+  readonly id: string;
+
+  /**
+   * The Id of the root Organizational Unit of the Organization
+   */
+  readonly rootId: string;
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    let orgProvider = new OrganizationProvider(this, "orgProvider");
+
+    let org = new CustomResource(this, "orgCustomResource", {
+      serviceToken: orgProvider.provider.serviceToken,
+      resourceType: "Custom::OrganizationCreation",
+    });
+
+    this.id = org.getAtt("OrganizationId").toString();
+
+    let root = new cr.AwsCustomResource(this, "RootCustomResource", {
+      onCreate: {
+        service: "Organizations",
+        action: "listRoots",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse("Roots.0.Id"),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      },
+      onUpdate: {
+        service: "Organizations",
+        action: "listRoots",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse("Roots.0.Id"),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      },
+      onDelete: {
+        service: "Organizations",
+        action: "listRoots",
+        physicalResourceId: cr.PhysicalResourceId.fromResponse("Roots.0.Id"),
+        region: "us-east-1", //AWS Organizations API are only available in us-east-1 for root actions
+      },
+      installLatestAwsSdk: false,
+      policy: cr.AwsCustomResourcePolicy.fromSdkCalls({
+        resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE,
+      }),
+    });
+
+    // Enabling SSM AWS Service access to be able to register delegated adminstrator
+    const enableSSMAWSServiceAccess = this.enableAWSServiceAccess("ssm.amazonaws.com");
+    const enableMultiAccountsSetupAWSServiceAccess = this.enableAWSServiceAccess("config-multiaccountsetup.amazonaws.com");
+
+    enableMultiAccountsSetupAWSServiceAccess.node.addDependency(org);
+    enableSSMAWSServiceAccess.node.addDependency(enableMultiAccountsSetupAWSServiceAccess);
+
+    //adding an explicit dependency as CloudFormation won't infer that calling listRoots must be done only when Organization creation is finished
+    //as there is no implicit dependency between the 2 custom resources
+    root.node.addDependency(org);
+
+    this.rootId = root.getResponseField("Roots.0.Id");
+  }
 
   private enableAWSServiceAccess(principal: string) {
-    const resourceName = principal==='ssm.amazonaws.com'?"EnableSSMAWSServiceAccess":"EnableMultiAccountsSetup";
-
-    return new cr.AwsCustomResource(this,
-      resourceName,
-      {
-        onCreate: {
-          service: 'Organizations',
-          action: 'enableAWSServiceAccess',
-          physicalResourceId: cr.PhysicalResourceId.of(resourceName),
-          region: 'us-east-1',
-          parameters: {
-            ServicePrincipal: principal,
-          }
+    const resourceName =
+      principal === "ssm.amazonaws.com"
```

Comment:
Would not rename the resource, since the docs state more services which the bootstrap kit will enable later on.

Reply:
Same comment as before, this logic should change.